### DISCUSSION TITLE
input help

### DISCUSSION THREAD

**Discussion by JacobTondreau** [autokey/autokey#960] [Q&A]
This is my code and I keep getting variations of errors depending on what I change, im basically a noob when it comes to coding so ChatGPT got me here but it is not working how I want it to. The furthest I could get it was to press "h" click and then it sat there. (if you are curious this is supposed to be an auto place and upgrade script for a tower in bloonsTD6)
any help is welcome
![fail](https://github.com/user-attachments/assets/b336e7d5-8d01-456b-ab99-8bf854fb2c8c)

import time
from pynput.keyboard import Controller as KeyboardController, Key
from pynput.mouse import Controller as MouseController, Button

# Create keyboard and mouse controllers
keyboard = KeyboardController()
mouse = MouseController()

# Step 1: Press the "h" key
keyboard.press('h')
keyboard.release('h')
time.sleep(0.1)  # Small delay

# Step 2: Click the left mouse button twice
mouse.click(Button.left, 1)
time.sleep(1)  # Small delay
mouse.click(Button.left, 1)
time.sleep(1)  # Small delay

# Step 3: Press the "1" key on the numpad four times
for _ in range(4):
#    keyboard.press(Key.num_lock)  # Ensure Num Lock is on
#    keyboard.release(Key.num_lock)
    keyboard.press(1)
    keyboard.release(1)
    time.sleep(0.2)  # Delay between presses

# Step 4: Press the "2" key on the numpad two times
for _ in range(2):
    keyboard.press(2)
    keyboard.release(2)
    time.sleep(0.2)  # Delay between presses

**Comment by josephj11** [comment]
Welcome to the AutoKey community, @JacobTondreau !

Another AI trainwreck. AFAIK, no AI has bothered to scrape AutoKey and our wiki. Or at least that's what the results look like.
IMHO, LLMs don't qualify as AI unless it means artificial ignorance! They don't actually "think" at all and just put words together using very fancy statistics and plagiarism.

Throw that "code" away and describe your script in English and write it in pseudocode if you are familiar with that and post it on [Gitter](https://gitter.im/autokey/autokey). Then we'll try to help you build something that actually works. The more details you provide about what you want to do and what application window(s) it will be manipulating, the faster we can create a solution.

If you're not sure how to do something, focus more on what the goal is and we can figure out the best way to accomplish it.

If you do post any pseudocode or other structured text, be sure to prefix it with a blank line followed by a line that only contains three backticks on it and follow the text with a line that only contains three backticks on it followed by a blank line. This markdown turns the text into a blockquote which preserves any formatting, especially the white space like indentation, making it much more readable. (The backtick is the key to the left of the number one on a standard qwerty keyboard.)
